eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -86,53 +86,69 @@ stages:
                     steps:
                       - checkout: self
 
+                      - pwsh: |
+                          $branchName = "azure-sdk-for-cpp-${{ artifact.Name }}-$(Build.BuildId)"
+                          if ('$(VcpkgForkBranchName)') { 
+                            Write-Host "Using queue time branch name" 
+                            $branchName = '$(VcpkgForkBranchName)'
+                          } 
+                          Write-Host "##vso[task.setvariable variable=PrBranchName]$branchName"
+                        displayName: Set fork branch name
+
                       - task: Powershell@2
                         inputs:
                           filePath: eng/scripts/Initialize-VcpkgRelease.ps1
                           arguments: >-
-                            -GitHubPat $(azuresdk-github-pat)
                             -GitHubRepo "$(Build.Repository.Name)"
                             -PackageSpecPath $(Pipeline.Workspace)/packages/${{artifact.Name}}/package-info.json
                             -SourceDirectory $(Pipeline.Workspace)/packages/${{artifact.Name}}/vcpkg
                             -Verbose
                           pwsh: true
                         displayName: Initialize vcpkg release
 
+                      # There are potential race conditions if this script runs
+                      # in parallel against the same branch name. Release only 
+                      # one package at a time.
                       - pwsh: |
+                          Write-Host "git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg"
                           git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg
                           if ($LASTEXITCODE -ne 0) {
                             Write-Error "Unable to check out vcpkg repo"
                             exit $LASTEXITCODE
                           }
+                          
+                          Write-Host "cd $(Pipeline.Workspace)/vcpkg"
                           cd $(Pipeline.Workspace)/vcpkg
 
-                          # There is no need to check out a branch, the
-                          # create-pull-request.yml template creates the branch
-
                           # Clean out the folder so that template files removed
                           # are not inadvertently re-added
-                          if (Test-Path "ports/${{ artifact.Name }}") {
-                            Remove-Item -v -r "ports/${{ artifact.Name }}"
+                          if (Test-Path "ports/${{ artifact.VcpkgPortName }}") {
+                            Remove-Item -v -r "ports/${{ artifact.VcpkgPortName }}"
```

Comment:
We seem to repeat `ports/${{ artifact.VcpkgPortName }}` a lot so it might be worth capturing that in a variable.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/1356